src/internal/events/events.go

```diff
@@ -159,3 +162,84 @@ func tenantHash(tenID string) string {
 	sum := md5.Sum([]byte(tenID))
 	return fmt.Sprintf("%x", sum)
 }
+
+// ---------------------------------------------------------------------------
+// metrics aggregation
+// ---------------------------------------------------------------------------
+
+type m string
+
+// metrics collection bucket
+const APICall m = "api_call"
+
+// configurations
+const (
+	reportInterval    = 1 * time.Minute
+	retentionDuration = 2 * time.Minute
+)
+
+// NewMetrics embeds a metrics bus into the provided context.  The bus can be
+// utilized with calls like Inc and Since.
+func NewMetrics(ctx context.Context, w io.Writer) (context.Context, func()) {
+	var (
+		// report interval time-bounds metrics into buckets.  Retention
+		// controls how long each interval sticks around.  Neither one controls
+		// logging rates; that's handled by dumpMetrics().
+		sink = metrics.NewInmemSink(reportInterval, retentionDuration)
+		cfg  = metrics.DefaultConfig("corso")
+		sig  = metrics.NewInmemSignal(sink, metrics.DefaultSignal, w)
+	)
+
+	cfg.EnableHostname = false
+	cfg.EnableRuntimeMetrics = false
+
+	gm, err := metrics.NewGlobal(cfg, sink)
+	if err != nil {
+		logger.CtxErr(ctx, err).Error("metrics bus constructor")
+		return ctx, func() { sig.Stop() }
+	}
+
+	stop := make(chan struct{})
+	go dumpMetrics(ctx, stop, sig)
+
+	flush := func() {
+		close(stop)
+		sig.Stop()
+		gm.Shutdown()
```

Comment:
should `gm.Shutdown()` be called before the other two so we get the final data it may have output?
```suggestion
		gm.Shutdown()
		close(stop)
		sig.Stop()
```

Reply:
I don't think that'll guarantee us any additional data.  But I _can_ tell the system to make signal print out a final response before we close.

Reply:
eh, idk, it just seems weird that we're stopping the thing that prints stuff before we stop the thing that makes the stuff we print. Maybe I'm just used to having to worry about ordering for stuff a lot though